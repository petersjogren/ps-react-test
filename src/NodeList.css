.nodelistitem {
  margin: 3px;
  color: #ddd;
  padding: 2px;
  font-size: 14px;
  border: 3px;
  border-radius: 10px;
  border-color: black;
  border-style: solid;
  background-color: green;
}
