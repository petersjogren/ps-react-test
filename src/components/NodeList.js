import React from "react";

export class NodeList extends React.Component {
  shouldComponentUpdate(newProps, newState) {
    if (this.props.templates.length !== newProps.templates.length) {
      return true;
    } else {
      return false;
    }
  }

  render() {
    return (
      <React.Fragment>
        <button onClick={this.props.onLoadDefault}>Load default</button>
        <button onClick={this.props.onLoadOther}>Load many</button>
        {this.props.templates.map((value, index) => {
          return (
            <div
              key={index}
              className="nodelistitem"
              draggable
              onDragStart={e => {
                //console.log(e.nativeEvent.offsetX, e.nativeEvent.offsetY);

                var payLoadString = JSON.stringify({
                  type: "CREATE_NODE",
                  templateIndex: index,
                  title: value.title
                });
                console.log("drag start", payLoadString);
                e.dataTransfer.setData("text/plain", payLoadString);
              }}
            >
              {value.title.substring(0, 20)}
            </div>
          );
        })}
      </React.Fragment>
    );
  }
}
