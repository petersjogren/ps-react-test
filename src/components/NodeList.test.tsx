import React from "react";
import ReactDOM from "react-dom";
import { NodeList } from "./NodeList";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <NodeList
      templates={[{ title: "One" }, { title: "Two" }]}
      onLoadDefault={() => {}}
      onLoadOther={() => {}}
      onCreateNode={(index, title) => {}}
    />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the same as last time", () => {
  const component = renderer.create(
    <NodeList
      templates={[{ title: "One" }, { title: "Two" }]}
      onLoadDefault={() => {}}
      onLoadOther={() => {}}
      onCreateNode={(index, title) => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
