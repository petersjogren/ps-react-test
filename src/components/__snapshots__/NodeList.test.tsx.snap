// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same as last time 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Load default
  </button>,
  <button
    onClick={[Function]}
  >
    Load many
  </button>,
  <div
    className="nodelistitem"
    draggable={true}
    onClick={[Function]}
    onDragStart={[Function]}
  >
    One
  </div>,
  <div
    className="nodelistitem"
    draggable={true}
    onClick={[Function]}
    onDragStart={[Function]}
  >
    Two
  </div>,
]
`;
